Foundation/Primitive/Block.hs

```diff
@@ -164,6 +167,14 @@ foldl' f initialAcc vec = loop 0 initialAcc
         | i .==# len = acc
         | otherwise  = loop (i+1) (f acc (unsafeIndex vec i))
 
+foldl1' :: PrimType ty => (ty -> ty -> ty) -> NonEmpty (Block ty) -> ty
+foldl1' f arr = let (initialAcc, rest) = splitAt 1 $ getNonEmpty arr
```

Comment:
`splitAt` is not cheap in Block, because Block don't support slicing, so that mean a full copy - 1 of the block.
I think it would be simpler just to unroll the loop directly and only work on indexes